### PR TITLE
Add default-true parameter "~enable_odom_tf"

### PR DESCRIPTION
**Pull request by bmagyar**
https://github.com/ros-controls/ros_controllers/issues/118

### CODE REVIEW

**Comment by adolfo-rt** [comment]
Could the test suite be expanded to make sure that with this setting no tf data is being published?

**Comment by bmagyar** [comment]
I'll cook something up and add to the PR.

**Comment by adolfo-rt** [comment]
Once that's done we can merge. Thanks for the quick response!

**Comment by bmagyar** [comment]
I added a new test to check that there is no frame when the tf odom is not enabled. I also added an extra test function to the general test to check that it is there in the default case.

File: diff_drive_controller/test/diff_drive_odom_tf_test.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+TEST_F(DiffDriveControllerTest, testNoOdomFrame)
+{
+  // wait for ROS
+  while(!isControllerAlive())
+  {
+    ros::Duration(0.1).sleep();
+  }
+  // set up tf listener
+  tf::TransformListener listener;
+  // wait for 5s
```
Comment (adolfo-rt) [inline diff_drive_controller/test/diff_drive_odom_tf_test.cpp]: The comment is out of sync with the line below (5s vs 2s). Maybe we can just remove it

File: diff_drive_controller/test/diff_drive_test.cpp [hunk 2]
```diff
@@ -121,6 +122,21 @@ TEST_F(DiffDriveControllerTest, testTurn)
...
+TEST_F(DiffDriveControllerTest, testOdomFrame)
+{
+  // wait for ROS
+  while(!isControllerAlive())
+  {
+    ros::Duration(0.1).sleep();
+  }
+  // set up tf listener
+  tf::TransformListener listener;
+  // wait for 5s
```
Comment (adolfo-rt) [inline diff_drive_controller/test/diff_drive_test.cpp]: Same here

**Comment by adolfo-rt** [comment]
Minor comments inline with the code.

File: diff_drive_controller/test/diff_drive_odom_tf_test.cpp (hunk 1 above)
Comment (bmagyar) [inline diff_drive_controller/test/diff_drive_odom_tf_test.cpp]: done

File: diff_drive_controller/test/diff_drive_test.cpp (hunk 2 above)
Comment (bmagyar) [inline diff_drive_controller/test/diff_drive_test.cpp]: done

**Comment by efernandez** [comment]
LGTM :+1:

**Comment by adolfo-rt** [comment]
+1. merging

**Comment by mikepurvis** [comment]
Thanks again!

**Comment by bmagyar** [comment]
:+1:

**Pull request merged**